hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -525,10 +545,15 @@ private boolean exposeInHmsProperties() {
     return maxHiveTablePropertySize > 0;
   }
 
-  private StorageDescriptor storageDescriptor(TableMetadata metadata, boolean hiveEngineEnabled) {
+  private StorageDescriptor storageDescriptor(
+      TableMetadata metadata, boolean hiveEngineEnabled, List<FieldSchema> oldCols) {
 
     final StorageDescriptor storageDescriptor = new StorageDescriptor();
-    storageDescriptor.setCols(HiveSchemaUtil.convert(metadata.schema()));
+    if (oldCols != null) {
+      storageDescriptor.setCols(oldCols);
```

Comment:
Maybe, safer just to do the convert but prune out the typeInfo like hive tables.